src/e2e/iOS/helpers/getSelection.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable fp/no-get-set */
+import { BrowserEnvironment } from '../../test-types'
+
+/** Returns a proxy getSelection object with async getters for selection properties. WebdriverIO cannot retrieve the entire selection object so we delegate to asynchronous getters. Top-level selection properties (e.g. .focusNode) are compatible with nested properties (e.g. .focusNode.textContent) are compatible with parent properties by defining ad hoc getters on promises.
+ *
+ * @example
+ * await getSelection().focusOffset
+ * await getSelection().focusNode
+ * await getSelection().focusNode?.textContent
+ *
+ **/
+const getSelection = (browser: BrowserEnvironment) => {
```

Comment:
Should this file be in iOS/helpers? It is also used by puppeteer.

Reply:
It can be used as-is with webdriverio, since `Browser<'async'>` is a `BrowserEnvironment`. However, the puppeteer helper should not have to know about `iOS` at all. I agree it would be better in a more generic directory.